src/Resources/contao/dca/tl_module.php

```diff
@@ -1053,4 +1057,29 @@ public function setMultiSrcFlags($varValue, DataContainer $dc)
 
 		return $varValue;
 	}
+
+
+	/**
+	 * Dynamically change attributes of the "pages" field
+	 *
+	 * @param mixed         $varValue
+	 * @param DataContainer $dc
+	 *
+	 * @return mixed
+	 */
+	public function setPagesFlags($varValue, DataContainer $dc)
+	{
+		if ($dc->activeRecord)
+		{
+			switch ($dc->activeRecord->type)
+			{
+				case 'search':
```

Comment:
This is unnecessarily complex. Why not the following?

```php
if ($dc->activeRecord && $dc->activeRecord->type == 'search')
{
    $GLOBALS['TL_DCA'][$dc->table]['fields'][$dc->field]['eval']['mandatory'] = false;
    unset($GLOBALS['TL_DCA'][$dc->table]['fields'][$dc->field]['eval']['orderField']);
}
```

BTW, is it really necessary to unset the order field?